src/Common/QuillLogger.h

```diff
@@ -0,0 +1,160 @@
+#pragma once
+
+#include <Common/QuillLogger_fwd.h>
+#include <Common/ThreadPool_fwd.h>
+
+#include <quill/core/Common.h>
+#include <quill/Logger.h>
+#include <quill/core/LogLevel.h>
+#include <quill/sinks/FileSink.h>
+#include <quill/sinks/StreamSink.h>
+#include <quill/Frontend.h>
+
+#include <deque>
+#include <filesystem>
+#include <future>
+
+using FreeThreadPool = ThreadPoolImpl<std::thread>;
+
+namespace DB
+{
+
+struct QuillFrontendOptions
+{
+    static constexpr quill::QueueType queue_type = quill::QueueType::UnboundedBlocking;
+
+    /**
+   * Initial capacity of the queue. Used for UnboundedBlocking, UnboundedDropping, and
+   * UnboundedUnlimited. Also serves as the capacity for BoundedBlocking and BoundedDropping.
+   */
+    static constexpr uint32_t initial_queue_capacity = 128 * 1024;
+
+    /**
+   * Interval for retrying when using BoundedBlocking or UnboundedBlocking.
+   * Applicable only when using BoundedBlocking or UnboundedBlocking.
+   */
+    static constexpr uint32_t blocking_queue_retry_interval_ns = 800;
+
+    /**
+   * Enables huge pages on the frontend queues to reduce TLB misses. Available only for Linux.
+   */
+    static constexpr quill::HugePagesPolicy huge_pages_policy = quill::HugePagesPolicy::Never;
```

Comment:
I wonder if we should rethink our approach to huge pages. I know we complain if transparent hugepages are enabled, but they should be fine with newer kernels and might be an improvement in many cases.

Reply:
I patched the lib to use global allocator instead of direct `mmap`. So this setting should be irrelevant.
I think there were some tests with huge pages in jemalloc but don't remember the conclusions.